civicrm.module

```diff
@@ -1029,13 +1029,13 @@ function civicrm_form_alter(&$form, &$form_state, $formID) {
  * @param $form_state
  */
 function civicrm_user_admin_permissions_submit($form, &$form_state) {
-  $rid = array_search('anonymous user', $form_state['values']['role_names']);
-  if ($rid === FALSE) {
+  $role = $form_state['values']['roles']['anonymous'];
+  if ($role === FALSE) {
```

Comment:
This doesn't look like a safe change to me. This condition **was** testing whether the results of `array_search` were `FALSE`. Not sure what it's testing now, but doesn't look right.